lib/internal/streams/pipe.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+const pipeline = require('internal/streams/pipeline');
+const Duplex = require('internal/streams/duplex');
+
+module.exports = function pipe(...streams) {
+  let cb;
+  let ret;
+
+  const r = pipeline(streams, function(err) {
+    if (cb) {
+      cb(err);
+    } else {
+      ret.destroy(err);
+    }
+  });
+  const w = streams[0];
+
+  ret = new Duplex({
+    writable: !!w?.writable,
+    readable: !!r?.readable,
+    objectMode: streams[0].readableObjectMode,
+    highWaterMark: 1
+  });
+
+  if (ret.writable) {
+    ret._write = function(chunk, encoding, callback) {
+      w.write(chunk, encoding, callback);
```

Comment:
I think this implementation is not correct as it will force highwatermark to 1 in the whole pipeline.

Reply:
Why do you think that?

Reply:
There is buffering in the wrapped streams.

Reply:
I see the problem now.